src/test/java/com/notably/logic/commands/NewCommandTest.java

```diff
@@ -0,0 +1,58 @@
+package com.notably.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.notably.commons.path.AbsolutePath;
+import com.notably.logic.commands.exceptions.CommandException;
+import com.notably.model.Model;
+import com.notably.model.ModelManager;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockImpl;
+import com.notably.model.block.BlockModel;
+import com.notably.model.block.BlockModelImpl;
+import com.notably.model.block.Title;
+import com.notably.model.suggestion.SuggestionModel;
+import com.notably.model.suggestion.SuggestionModelImpl;
+import com.notably.model.viewstate.ViewStateModel;
+import com.notably.model.viewstate.ViewStateModelImpl;
+
+class NewCommandTest {
+    private static Model model;
+
+    @BeforeEach
+    public void setUp() {
+        // Set up model
+        BlockModel blockModel = new BlockModelImpl();
+        SuggestionModel suggestionModel = new SuggestionModelImpl();
+        ViewStateModel viewStateModel = new ViewStateModelImpl();
+        model = new ModelManager(blockModel, suggestionModel, viewStateModel);
+
+    }
+    @Test
+    void execute_validInput_updatedDatastructure() {
+        final Block currentBlock = (new BlockImpl(new Title("CS2103")));
+
+        final AbsolutePath expectedPath = AbsolutePath.fromString("/CS2103");
+
+        model.addBlockToCurrentPath(currentBlock);
+
+        assertTrue(model.hasPath(expectedPath));
+    }
+
+    @Test
+    void execute_invalidInputDuplicateTitle_throwsDuplicateBlockException() throws CommandException {
```

Comment:
Since we restrict the note title to be alphanumeric, `invalidInput` means that the title contains a non-alphanumeric. So the test cases that we need to consider:
- valid input (alphanumeric) + not dup title: `execute_validInputNotDuplicateTitle_updatedDataStructure`
- valid input + dup title: `execute_validInputDuplicateTitle_throws...`
- invalid input + not dup title: `execute_invalidInputNotDuplicateTitle_throws...`
(invalid input + dup title might be redundant).

Since your `invalidInputDuplicateTitle` test case is for valid input + dup title, you may want to add the case for invalid input + not dup title.

Reply:
For `newCommandTest` i was thinking that it would only handle CommandException(RunTime Exception). Since alphanumeric variable is being checked at the parser. I've left it at the `NewCommandParserTest`. Let me know what you think @johannagwan .

Reply:
Oh I see, sure then!